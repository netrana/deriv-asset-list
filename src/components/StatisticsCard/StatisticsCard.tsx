import React, { FC } from "react";
import classNames from "classnames";

import { Props } from "./types";
import styles from "./statisticsCard.module.scss";
import NXCard from "components/NXCard";

const StatisticsCard: FC<Props> = (props) => {
  const { title, children, className = "", isActive } = props;
  return (
    <NXCard
      metaTitle={title}
      cssClassNames={classNames({
        [styles.statisticsCard]: true,
        [styles.active]: isActive,
        [className]: Boolean(className),
      })}
    >
      {children}
    </NXCard>
  );
};

export default StatisticsCard;
