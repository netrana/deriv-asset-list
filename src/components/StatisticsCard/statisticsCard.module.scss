@import 'styles/globals.scss';

$secondaryTextColor: rgba(0, 0, 0, 0.45);
$primaryTextColor: $primaryThemeColor;

.statisticsCard {
  color: $secondaryTextColor;
  margin-bottom: 1rem;

  :global(.ant-card-body) {
    border-radius: $borderRadius;
    padding: 24px 24px 10px 24px;
    :global(.ant-card-meta-title) {
      color: rgba(0,0,0,.45)
    }    
  }

  :global(.ant-card-meta) {
    padding: 0px 10px 10px 0px;
  } 
}

.active {   
  :global(.ant-card-body) {
    background-image: $backgroundImage;     
    * {
      color: white !important;  
    } 
  }
}
