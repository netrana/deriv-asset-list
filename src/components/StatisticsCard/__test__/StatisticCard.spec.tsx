import React from "react";
import render from "react-test-renderer";

import StatisticsCard from "../StatisticsCard";

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  FC: jest.fn(),
}));

jest.mock("antd", () => ({
  ...jest.requireActual("antd"),
  Card: "Card",
}));

describe("StatisticsCard component test", () => {
  describe("snapshot test", () => {
    it("should match snapshot with default props", () => {
      const tree = render
        .create(<StatisticsCard title="Test Title" />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it("should match snapshot with isActive true", () => {
      const tree = render
        .create(<StatisticsCard title="Test Title" isActive />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
