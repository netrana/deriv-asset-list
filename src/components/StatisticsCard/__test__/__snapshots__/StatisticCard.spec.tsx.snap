// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticsCard component test snapshot test should match snapshot with default props 1`] = `
<Card
  className="dashCard"
>
  <p>
    Test Title
  </p>
</Card>
`;

exports[`StatisticsCard component test snapshot test should match snapshot with isActive true 1`] = `
<Card
  className="dashCard active"
>
  <p>
    Test Title
  </p>
</Card>
`;
